### PR TITLE
LPS-180955 Create integration tests for LPS-163716

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPS-180955](https://issues.liferay.com/browse/LPS-180955)

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 1]
```diff
@@ -1039,6 +991,10 @@ public void testObjectFieldSettings() throws Exception {
 
 		_objectDefinitionLocalService.deleteObjectDefinition(
 			objectDefinition.getObjectDefinitionId());
+
+		// Default value
+
+		_testObjectFieldDefaultValue();
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:997]: Moreover, you can change the _objectField_ [here](https://github.com/liferay/liferay-portal/blob/95d934c3b3e3e0d5db4917039f1e8e2a4e0b12c6/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java#L1012) to name with "text". We can use this field to test expressionBuilder.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 2]
```diff
@@ -1393,6 +1406,273 @@ private String _testAddSystemObjectField(
 		return objectDefinitionName;
 	}
 
+	private void _testObjectFieldDefaultValue() throws Exception {
+		ObjectDefinition objectDefinition =
```
Comment (murilostodolni) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1410]: @devpedrotavares this objectDefinition is no longer necessary.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 3]
```diff
@@ -1393,6 +1406,273 @@ private String _testAddSystemObjectField(
...
+
+		_assertObjectFieldSettingDefaultValue(
+			_listTypeEntryKey, ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE,
+			objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		_objectFieldLocalService.deleteObjectField(objectField);
+
+		// Try to add field as state without default value
```
Comment (murilostodolni) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1556]: @devpedrotavares this part related to required values you can put in this part [here](https://github.com/liferay/liferay-portal/blob/95d934c3b3e3e0d5db4917039f1e8e2a4e0b12c6/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java#L844). Just below this [line](https://github.com/liferay/liferay-portal/blob/95d934c3b3e3e0d5db4917039f1e8e2a4e0b12c6/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java#L885). You can use the same pattern that already exist.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 4]
```diff
@@ -1393,6 +1406,273 @@ private String _testAddSystemObjectField(
...
+		objectField.setObjectFieldSettings(
+			Arrays.asList(
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+					expressionBuilderDefaultValue),
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+					ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER)));
+
+		objectField = _updateCustomObjectField(objectField);
```
Comment (murilostodolni) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1512]: @devpedrotavares you can add `objectField.setObjectFieldSettings()` as parameter in `_updateCustomObjectField`.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 5]
```diff
@@ -1393,6 +1406,273 @@ private String _testAddSystemObjectField(
...
+						ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+						"able", "able", false)));
+
+		_objectDefinitionLocalService.publishCustomObjectDefinition(
+			TestPropsValues.getUserId(),
+			objectDefinition.getObjectDefinitionId());
+
+		// Add field with default value input as value
+
+		ObjectField objectField = _addCustomObjectField(
```
Comment (murilostodolni) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1425]: @devpedrotavares when you take this code to the other method, change `objectFiled` to `picklistObjectField`.

**Review by murilostodolni — CHANGES_REQUESTED** [review]
Hi @devpedrotavares 

I finish my review. Fill free to call me if have any questions. Thank you

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java (hunk 4 above)
Comment (devpedrotavares) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java]: Hey @murilostodolni, don't you think we should have the `_updateCustomObjectField` be generic?

Comment (murilostodolni) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java]: Yes, the `_updateCustomObjectField` should be generic. You can do something like this:

```
picklistObjectField = _updateCustomObjectField(picklistObjectField.setObjectFieldSettings(
			Arrays.asList(
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
					_listTypeEntryKey),
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
					ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))));
```

Comment (murilostodolni) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java]: You don't need change the method, just inline.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 6]
```diff
@@ -884,6 +842,93 @@ public void testObjectFieldSettings() throws Exception {
...
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						_listTypeEntryKey),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER)));
+
+			Assert.fail();
+		}
+		catch (ObjectFieldSettingValueException.InvalidValue
+					objectFieldSettingValueException) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:888]: @devpedrotavares can you sort the exceptions alphabetically? `ObjectFieldSettingValueException.InvalidValue` before `ObjectFieldSettingValueException.MissingRequiredValues`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 7]
```diff
@@ -884,6 +842,93 @@ public void testObjectFieldSettings() throws Exception {
...
+				objectFieldSettingValueException.getMessage(),
+				StringBundler.concat(
+					"The value ",
+					ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER,
+					" of setting ",
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+					" is invalid for object field picklist"));
+		}
+
+		String randomString = RandomTestUtil.randomString();
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:900]: @devpedrotavares I would rename it to `defaultValueObjectFieldSettingValue`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 8]
```diff
@@ -1037,8 +1119,105 @@ public void testObjectFieldSettings() throws Exception {
 		_assertObjectFieldSetting(
 			"showCounter", objectField.getObjectFieldId(), "false");
 
-		_objectDefinitionLocalService.deleteObjectDefinition(
+		// Default value
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1122]: @devpedrotavares can you change this comment to `// Business type picklist` to follow the same existing pattern? 

And I believe we should place it before the `// Business type text`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 9]
```diff
@@ -1263,6 +1442,36 @@ public void testUpdateRequired() throws Exception {
...
+	private void _assertObjectEntryDefaultValue(
+			String expectedDefaultValue, ObjectField objectField,
+			Map<String, Serializable> values)
+		throws Exception {
+
+		ObjectEntry objectEntry = _objectEntryLocalService.addObjectEntry(
+			TestPropsValues.getUserId(), 0, objectField.getObjectDefinitionId(),
+			values, ServiceContextTestUtil.getServiceContext());
+
+		Map<String, Serializable> objectEntryValues = objectEntry.getValues();
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1469]: @devpedrotavares can we keep the `values` name without having to initialize another map?

```
values = objectEntry.getValues();
```

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 10]
```diff
@@ -1037,8 +1119,105 @@ public void testObjectFieldSettings() throws Exception {
...
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+			objectField.getObjectFieldId(), _listTypeEntryKey);
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+			objectField.getObjectFieldId(),
+			ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE);
+
+		_assertObjectEntryDefaultValue(
+			_listTypeEntryKey, objectField, new HashMap<>());
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1216]: @devpedrotavares this code looks duplicated.. Could you reuse code by creating a private method which will receive the  `required` and `state` parameters? Something like:

```
private ObjectField _addPicklistObjectField(
	ObjectDefinition objectDefinition, boolean required, boolean state) {

	ObjectField objectField = _addCustomObjectField(
		new PicklistObjectFieldBuilder(
		).labelMap(
			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
		).listTypeDefinitionId(
			_listTypeDefinition.getListTypeDefinitionId()
		).name(
			"a" + RandomTestUtil.randomString()
		).objectDefinitionId(
			objectDefinition.getObjectDefinitionId()
		).objectFieldSettings(
			Arrays.asList(
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
					_listTypeEntryKey),
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
					ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))
		).required(
			required
		).state(
			state
		).build());

	_assertObjectFieldSetting(
		ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
		objectField.getObjectFieldId(), _listTypeEntryKey);
	_assertObjectFieldSetting(
		ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
		objectField.getObjectFieldId(),
		ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE);
	
	return objectField;
}
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 11]
```diff
@@ -1393,6 +1406,273 @@ private String _testAddSystemObjectField(
...
+			objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		_objectFieldLocalService.deleteObjectField(objectField);
+
+		// Add field with default value as expression builder
+
+		String expressionBuilderDefaultValue = String.format(
+			"\"%s\"", _listTypeEntryKey);
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java]: Great job on this, @murilostodolni! 🎉

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:forward

**Comment by devpedrotavares** [comment]
ci:forward:force

**Pull request closed without merging**